### PR TITLE
[#136226] prevent deletion of order form for purchased orders

### PR DESCRIPTION
**Pull request by meara**
- https://pm.tablexi.com/issues/136226

### CODE REVIEW

File: app/controllers/order_detail_stored_files_controller.rb [hunk 1]
```diff
@@ -48,19 +48,23 @@ def upload_order_file
...
       render :order_file
     end
   end
 
   # GET /orders/:order_id/order_details/:order_detail_id/remove_order_file
   def remove_order_file
-    if @order_detail.stored_files.template_result.all?(&:destroy)
-      flash[:notice] = "The uploaded Order File has been deleted successfully" # TODO: I18n
+    if @order_detail.new?
+      if @order_detail.stored_files.template_result.all?(&:destroy)
```
Comment (jhanggi) [inline app/controllers/order_detail_stored_files_controller.rb]: Will `@order_detail.stored_files.template_result.destroy_all` do the same thing? I forget what it returns.

File: app/controllers/order_detail_stored_files_controller.rb [hunk 2]
```diff
@@ -48,19 +48,23 @@ def upload_order_file
...
+        flash[:notice] = text('remove_order_file.success')
```
Comment (jhanggi) [inline app/controllers/order_detail_stored_files_controller.rb]: Use double quotes.

Comment (meara) [inline app/controllers/order_detail_stored_files_controller.rb]: Huh it is weird that rubocop didn't flag on that...

File: app/controllers/order_detail_stored_files_controller.rb (hunk 1 above)
Comment (meara) [inline app/controllers/order_detail_stored_files_controller.rb]: It returns the collection of objects that it destroys... idk if that is quite what we we want for that boolean.

https://apidock.com/rails/ActiveRecord/Relation/destroy_all

Comment (jhanggi) [inline app/controllers/order_detail_stored_files_controller.rb]: I think it's fine to leave as is. Although, technically `destroy` also returns the single object that was just destroyed.

I think it's probably fine to leave as is. I can't think of where this wouldn't work. And really there's only ever one of these files at a time.

File: app/controllers/order_detail_stored_files_controller.rb (hunk 2 above)
Comment (meara) [inline app/controllers/order_detail_stored_files_controller.rb]: Okay, I am genuinely mystified by this. I can't find anything in the rubocop config that explains why it should not have flagged on this.

**Pull request merged**